isovar/protein_sequence.py

```diff
@@ -304,4 +340,122 @@ def ascending_sort_key(self):
             -self.num_mismatches_before_variant,
             -self.num_mismatches_after_variant,
             len(self.amino_acids),
-        )
\ No newline at end of file
+        )
+
+    @property
+    def num_mutant_amino_acids(self):
+        """
+        Number of mutant amino acids contained in this sequence.
+
+        Returns
+        -------
+        int
+        """
+        return self.mutation_end_idx - self.mutation_start_idx
+
+    @property
+    def contains_deletion(self):
+        """
+        Returns True if this protein sequence contains the residues before
+        and after a deletion.
+
+        Returns
+        -------
+        bool
+        """
+        #
+        # Deletions are represented by a mutation interval with the same start/end
+        # but to have some sequence before and after the start/end have to be greater
+        # than 0 and less than the length of the sequence
+        # Example:
+        #
+        #    -0-1-2-3-4-5-6-7
+        #    -S-I-I-N-F-E-K-L
+        #
+        #    ^ start/end = 0, only see residue after
+        #    ----^ start/end = 2, deletion occured between two 'I' residues
+        #    ----------------^ start/end = 8, only see residue before
+        #
+        return (
+            self.contains_mutation and
+            (self.num_mutant_amino_acids == 0) and
+            (0 < self.mutation_start_idx < len(self))
```

Comment:
when would `mutation_start_idx` be <=0?

Reply:
When the mutation starts at the first amino acid then `mutation_start_idx == 0`. For deletions this actually means the deletion is *before* the start of the amino acid sequence, in which case I don't think it should count.